File: applications/account-rest-service/src/main/java/com/vmware/spring/geode/showcase/account/controller/AccountController.java
package com.vmware.spring.geode.showcase.account.controller;

import com.vmware.spring.geode.showcase.account.domain.account.Account;
import com.vmware.spring.geode.showcase.account.repostories.AccountRepository;
import lombok.AllArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.util.Optional;

@AllArgsConstructor
@RestController
public class AccountController
{
    private final AccountRepository accountRepository;

    @PostMapping("accounts")
    public <S extends Account> S save(@RequestBody S account)
    {
        return accountRepository.save(account);
    }

    @GetMapping("accounts/{id}")
    public Optional<Account> findById(@PathVariable String id)
    {
        return accountRepository.findById(id);
    }

    @DeleteMapping("accounts/{id}")
    public void deleteById(@PathVariable String id)
    {
        accountRepository.deleteById(id);
    }
}

File: applications/account-jdbc-caching-rest-service/src/main/kotlin/com/vmware/spring/geode/showcase/account/controller/AccountController.kt
package com.vmware.spring.geode.showcase.account.controller

import com.vmware.spring.geode.showcase.account.domain.account.Account
import com.vmware.spring.geode.showcase.account.service.AccountService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

/**
 * @author Gregory Green
 */
@RestController
class AccountController(private val accountService: AccountService) {

    @PostMapping("accounts")
    fun save(@RequestBody account: Account) {
        accountService.save(account)
    }

    @GetMapping("accounts/{id}")
    fun findByAccountId(@PathVariable("id") id: String): ResponseEntity<Account> {
        var account: Account? = accountService.findByAccountId(id) ?: return ResponseEntity.notFound().build()
        return ResponseEntity.ok(account)
    }
}